OasisDEX/oasis-borrow
# Follow vault list on my page

# Follow vault list on my page

Covers:
https://app.shortcut.com/oazo-apps/story/5886/follow-vaults-overview-section-ui-component
https://app.shortcut.com/oazo-apps/story/5891/followed-vaults-list-view-vault
and a little bit more

**PLEASE READ through changes list, how to test and additional notes before testing.**

This PR adds skeleton preloader ONLY for follow vault. User's positions at this moment doesn't have preloaders at all and are still displayed in old layout - not in discovery table. This is for next PR as since entire follow vault feature is hidden behind feature flag, temporary inconstancy won't matter.

![image](https://user-images.githubusercontent.com/16230404/213296750-18157bb2-1c11-4610-89cc-2987dd5a16b2.png)
![image](https://user-images.githubusercontent.com/16230404/213296611-9ecf0c98-8cbd-4b5a-80d0-e9bfd8415478.png)
![image](https://user-images.githubusercontent.com/16230404/213296957-fd9202ff-4bf4-473f-865a-9bc1c23bde00.png)
  
## Changes 👷‍♀️

- Huge Discover table overhaul allowing it to be more flexible and accepting more data than just discover,
- created api handler and overvable for loading user's followed vaults,
- created table with list of followed vaults for owner's page,
- created first version of skeleton preloader compontent to be used in other places - probably need to be expanded a little.
  
## How to test 🧪

- Go to owner's page (yours or somebody else's) and see "Follow" panel.
- see if preloader works as intended,
- see if empty state works as intended,
- see if list is populated with vaults that you decided to follow.

DO NOT report issues with following or unfollowing, this PR is only about list on owner's page.

## Additional notes and know bugs 🗒️

- You won't be able to unfollow vault until is merged: https://github.com/OasisDEX/oasis-borrow/pull/1861 is merged,
- sometimes during loading there is random circle preloaded on skeleton - this is a bug from component below with product cards and will be addressed with adding skeleton preloaders to the rest of that page,
- some header labels in table might be different than in design, but the names were unified to the ones from Discover - if they are going to be changed it's a matter of different discussion,
- guni vault shows incorrect data for PnL and liquidity, last 7 days yield is unavailable at all - this is outside of scope for this PR.
